src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTests.cs

```diff
@@ -627,6 +627,14 @@ .maxstack  1
         [Fact, WorkItem(540019, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/540019")]
         public void TestBug6156()
         {
+            // Note: this test originally required a mismatch between the compile-time view of what the
+            // program should do and the runtime view.  The runtime considers `ref` and `out` to be the
+            // same, so it had a different view of what one method was overriding than C# did.  However,
+            // in the context of implementing the covariant return types feature, we added code in the compiler to
+            // cause the compiler to emit methodimpl records whenever there is a mismatch.  As a consequence
+            // of that, we updated this test to now require that the runtime behavior and the compile-time behavior
+            // of this program match, so that at runtime it obeys the behavior specified by the C#
+            // language specification.
```

Comment:
Interesting... we should note this breaking change somewhere

Reply:
Looks more like a bug fix than a breaking change. We generated code to call the wrong (unrelated!) method before. Now the code causes the right method to be called.

---
In reply to: [437698623](https://github.com/dotnet/roslyn/pull/44025#discussion_r437698623) [](ancestors = 437698623)

Reply:
Depends on how old this behavior was. It's certainly a change and it could be breaking.